superset-frontend/src/dashboard/components/IconButton.tsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { MouseEventHandler } from 'react';
+import Icons from 'src/components/Icons';
+import { SupersetTheme } from '@superset-ui/core';
+import * as AntdIcons from '@ant-design/icons/lib/icons';
+
+interface IconButtonProps {
+  icon: keyof typeof Icons | keyof typeof AntdIcons;
```

Comment:
I think we can declare this somewhere in the Icons directory so that it can be re-used without having to redefine it all the times

Reply:
Yes. This is part of improving the type safety of `Icons`. The same code will also prevent something like `<Icons.UnknownIcon>`. I think that we can change this after the improvement.